eclair-core/src/main/scala/fr/acinq/eclair/channel/ErrorHandlers.scala

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright 2022 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.channel
+
+import akka.actor.typed.scaladsl.adapter.actorRefAdapter
+import akka.actor.{ActorRef, FSM}
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin.{ByteVector32, OutPoint, SatoshiLong, Transaction}
+import fr.acinq.eclair.NotificationsLogger.NotifyNodeOperator
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher.{WatchOutputSpent, WatchTxConfirmed}
+import fr.acinq.eclair.channel.Channel.UnhandledExceptionStrategy
+import fr.acinq.eclair.channel.Helpers.Closing
+import fr.acinq.eclair.channel.publish.TxPublisher
+import fr.acinq.eclair.channel.publish.TxPublisher.{PublishFinalTx, PublishReplaceableTx, PublishTx}
+import fr.acinq.eclair.crypto.keymanager.ChannelKeyManager
+import fr.acinq.eclair.transactions.Transactions
+import fr.acinq.eclair.transactions.Transactions.ClosingTx
+import fr.acinq.eclair.wire.protocol.{AcceptChannel, ChannelReestablish, Error, OpenChannel}
+import fr.acinq.eclair.{NodeParams, NotificationsLogger}
+
+import java.sql.SQLException
+
+/**
+ * Created by t-bast on 28/03/2022.
+ */
+
+/**
+ * This trait contains handlers for error scenarios (channel closing, force-closing, unhandled, exceptions, etc).
+ */
+trait ErrorHandlers extends FSM[ChannelState, ChannelData] with UtilityHandlers {
+
+  // @formatter:off
+  def nodeParams: NodeParams
+  def remoteNodeId: PublicKey
+  def keyManager: ChannelKeyManager
+
+  def txPublisher: akka.actor.typed.ActorRef[TxPublisher.Command]
+  def blockchain: akka.actor.typed.ActorRef[ZmqWatcher.Command]
+  def peer: akka.actor.ActorRef
+  def activeConnection: akka.actor.ActorRef
+  def self: akka.actor.ActorRef
+  // @formatter:on
+
+  implicit def implicitLog: akka.event.DiagnosticLoggingAdapter
```

Comment:
We should probably move those either in a `UtilityHandlers` or in an other adhoc trait.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2217/commits/b18bd74e935dfed75c1692157784ee63aa3eebf4

Reply:
I have an alternative commit c9fa7bd2d37e937b883207aea572692537211d17 using the [cake pattern](https://medium.com/rahasak/scala-cake-pattern-e0cd894dae4e). In general I'm not a fan of this pattern, but here the traits are really not meant to be extended by other than `Channel`, and it saves us from duplicating the args altogether.

The move `val` -> `protected val` is optional.

Reply:
Nice, I agree, this is the perfect place for the cake pattern. Done in https://github.com/ACINQ/eclair/pull/2217/commits/7f2c4af930c62e675e04a37f61efb3775dc65128